model/project_parser_test.go

```diff
@@ -1411,6 +1411,43 @@ buildvariants:
 			checkIsTaskGroupTaskUnit(t, bvtu)
 		}
 	})
+
+	t.Run("MaxHostsForTaskGroup", func(t *testing.T) {
+		validMaxHostYml := `
+tasks:
+- name: example_task_1
+- name: example_task_2
+task_groups:
+- name: example_task_group
+  max_hosts: -1
+  tasks:
+  - example_task_1
+  - example_task_2
+buildvariants:
+- name: bv
+  display_name: "bv_display"
+  tasks:
+  - name: example_task_group
+  - name: inline_task_group
+    task_group:
+      max_hosts: -1
+      tasks:
+      - example_task_1
+      - example_task_2
```

Comment:
Would be slightly preferable to test with a different task group definition than the one for `example_task_group` so we know that `max_hosts` is set based on the tasks define inline rather than the `example_task_group` tasks.